### PR TITLE
Fix RecursionError: Limit depth of nested Objects when generating fuzz parameters

### PR DESCRIPTION
**Pull request by Chandra158**
### Overview

```sh
...
...
  File "/github/fuzz-lightyear/fuzz_lightyear/fuzzer.py", line 81, in _fuzz_parameter
    strategy = fuzz_fn(parameter, operation_id, required=required)  # type: ignore
  File "/github/fuzz-lightyear/fuzz_lightyear/fuzzer.py", line 178, in _fuzz_array
    elements=_fuzz_parameter(item, operation_id, required=required),
  File "/github/fuzz-lightyear/fuzz_lightyear/fuzzer.py", line 81, in _fuzz_parameter
    strategy = fuzz_fn(parameter, operation_id, required=required)  # type: ignore
  File "/github/fuzz-lightyear/fuzz_lightyear/fuzzer.py", line 213, in _fuzz_object
    if strategy:
...
...
packages/hypothesis/strategies/_internal/lazy.py", line 139, in __repr__
    sig = signature(self.function)
  File "/usr/lib/python3.8/inspect.py", line 3105, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
...
...
  File "/usr/lib/python3.8/enum.py", line 339, in __call__
    return cls.__new__(cls, value)
RecursionError: maximum recursion depth exceeded while calling a Python object
```

- `APPSEC-5131`
- For self-referencing Objects, generating fuzz parameters results in a non-ending recursive call
- In this PR:
   - fixing the nested object depth to a max limit 
   - this limit ( `--depth`) can be passed as an argument (`default=6`)

### CODE REVIEW

File: fuzz_lightyear/usage.py [hunk 1]
```diff
@@ -104,6 +104,13 @@ def parse_args(argv: Optional[List[str]] = None) -> argparse.Namespace:
         ),
     )
 
+    parser.add_argument(
+        '--depth',
+        default=6,
```
Comment (jpdakran) [inline fuzz_lightyear/usage.py:109]: Is there a reason for defaulting this to 6?

File: fuzz_lightyear/main.py [hunk 2]
```diff
@@ -121,12 +124,14 @@ def run_tests(
...
     :param ignore_exceptions: if True, ignores HTTP exceptions to requests.
     :param disable_unicode: if True, only use ASCII characters to fuzz strings
+    :param max_fuzz_depth: used for preventing recursion Objects with self reference
     """
     if seed is not None:
         get_settings().seed = seed
     if disable_unicode:
         get_settings().unicode_enabled = False
 
+    get_settings().max_fuzz_depth = max_fuzz_depth
```
Comment (jpdakran) [inline fuzz_lightyear/main.py:134]: Do we need to check if this is already set here to not overwrite it?

Comment (Chandra158) [reply fuzz_lightyear/main.py:134]: At this point `max_fuzz_depth` will either have the value passed by user as argument or the default value. For the latter case, it'll have no effect as it'll set it to the default value.

File: fuzz_lightyear/usage.py (hunk 1 above)
Comment (Chandra158) [reply fuzz_lightyear/usage.py:109]: No specific reason. Just a value from what I've seen in generic request payloads. Was not too sure of this; so provided a way to pass it as argument.

**Review by jpdakran — APPROVED** [review]

File: fuzz_lightyear/main.py [hunk 3]
```diff
@@ -121,12 +124,14 @@ def run_tests(
     :param seed: used for random generation of test input
     :param ignore_exceptions: if True, ignores HTTP exceptions to requests.
     :param disable_unicode: if True, only use ASCII characters to fuzz strings
+    :param max_fuzz_depth: used for preventing recursion Objects with self reference
     """
     if seed is not None:
         get_settings().seed = seed
```
Comment (lorenzodb1) [inline fuzz_lightyear/main.py:130]: Does this line (and the other lines where `get_settings` is called) actually do anything? `get_settings` returns `Settings()`, so my understanding here is that this line sets the value of `seed` in `Settings`, but then that `Settings` object gets lost (i.e. the `Settings` returned on L134 would not have the same value for `seed` set in here).

Comment (Chandra158) [reply fuzz_lightyear/main.py:130]: @lorenzodb1 The `Settings` class is singleton here ([`@lru_cache(maxsize=1)`](https://github.com/Yelp/fuzz-lightyear/blob/master/fuzz_lightyear/settings.py#L24)). So, the values set here will persist and will be used in other parts of the project.

**Pull request merged**